Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
I guess the same pattern is used for both perl and ruby but they don't have elaborate DSLs (yet) that would benefit at this point.

So one question is whether to also include dependency-less and requirement-less `without-python` and `with-python` options as cases for needs_python?("python") to return true.